### ISSUE TITLE
Squiz.Arrays.ArrayDeclaration false positive for static functions

### ISSUE THREAD

**Issue by morozov** [squizlabs/PHP_CodeSniffer#3060]
**Describe the bug**
A false-positive rule violation is detected that prevents auto-fixing.

**Code sample**
```php
<?php

function test()
{
    yield [
        static function (): void {
        },
    ];
}
```

**Custom ruleset**
None.

**To reproduce**
1. Run `phpcs -s --standard=Squiz --sniffs=Squiz.Arrays.ArrayDeclaration,Squiz.WhiteSpace.ScopeKeywordSpacing test.php`
```
FILE: /home/morozov/Projects/phpcs-playground/test.php
------------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
------------------------------------------------------------------------------------------------
 6 | ERROR | [x] The first value in a multi-value array must be on a new line
   |       |     (Squiz.Arrays.ArrayDeclaration.ValueNoNewline)
------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
------------------------------------------------------------------------------------------------

Time: 44ms; Memory: 8MB
```
2. Run `phpcbf --standard=Squiz --sniffs=Squiz.Arrays.ArrayDeclaration,Squiz.WhiteSpace.ScopeKeywordSpacing test.php`
```
PHPCBF RESULT SUMMARY
----------------------------------------------------------------------
FILE                                                  FIXED  REMAINING
----------------------------------------------------------------------
/home/morozov/Projects/phpcs-playground/test.php      FAILED TO FIX
----------------------------------------------------------------------
A TOTAL OF 0 ERRORS WERE FIXED IN 1 FILE
----------------------------------------------------------------------
PHPCBF FAILED TO FIX 1 FILE
----------------------------------------------------------------------

Time: 95ms; Memory: 8MB
```

**Expected behavior**
No error is reported.

**Versions (please complete the following information):**
 - OS: Linux
 - PHP: 7.4.9
 - PHPCS: 3.5.6
 - Standard: Squiz
